app/src/main/java/com/example/plannr/DisplayCoursesFragment.java

```diff
@@ -0,0 +1,232 @@
+                container, false);
+        binding = FragmentDisplayCoursesBinding.inflate(inflater, container, false);
+
+        pullData();
+
+        return binding.getRoot();
+    }
+
+    private Course[] getData() {
+        Course[] courses = new Course[CourseRepository.getCourses().size()];
+
+        for (int i = 0; i < CourseRepository.getCourses().size(); i++) {
+            courses[i] = CourseRepository.getCourses().get(i);
+        }
+        return courses;
+    }
+
+    public void pullData() {
+        db.ref.child("offerings").get().addOnCompleteListener(new OnCompleteListener<DataSnapshot>() {
+            @Override
+            public void onComplete(@NonNull Task<DataSnapshot> task) {
+                if (!task.isSuccessful()) {
+                    Log.e("firebase", "Error getting data", task.getException());
+                } else {
+                    Log.d("firebase", String.valueOf(task.getResult().getValue()));
+                    commitData((Map<String, Object>) task.getResult().getValue());
+                }
+            }
+        });
+    }
+
+    public void commitData(Map<String, Object> courses) {
+        CourseRepository repository = CourseRepository.getInstance();
+
+        for (Map.Entry<String, Object> entry : courses.entrySet()) {
+            String name = ((Map) entry.getValue()).get("courseName").toString();
+            String code = ((Map) entry.getValue()).get("courseCode").toString();
+            String prerequisites = ((Map) entry.getValue()).get("prerequisites").toString();
+            boolean fall = ((Map) entry.getValue()).get("fallAvailability").equals("true");
+            boolean summer = ((Map) entry.getValue()).get("summerAvailability").equals("true");
+            boolean winter = ((Map) entry.getValue()).get("winterAvailability").equals("true");
+            int id = Integer.parseInt(entry.getKey());
+
+            Course temp = new Course(code, name, fall, summer, winter, prerequisites, id);
+
+            repository.addCourse(temp);
+        }
+
+        generatePage();
+    }
+
+    @SuppressLint("ClickableViewAccessibility")
+    private void generatePage() {
+        Course[] courses = getData();
+
+        LinearLayout.LayoutParams layoutParams = new LinearLayout.LayoutParams(
+                LinearLayout.LayoutParams.MATCH_PARENT,
+                LinearLayout.LayoutParams.WRAP_CONTENT,
+                100.0f);
+        LinearLayout.LayoutParams buttonParams = new LinearLayout.LayoutParams(
+                LinearLayout.LayoutParams.MATCH_PARENT,
+                LinearLayout.LayoutParams.WRAP_CONTENT);
+
+        LinearLayout page = new LinearLayout((getContext()));
+        page.setOrientation(LinearLayout.VERTICAL);
+        ScrollView scroll = new ScrollView(getContext());
+
+        Button addCourse = new Button(getContext());
+        addCourse.setText("Add Course");
+        addCourse.setTextSize(20);
```

Comment:
shouldn't this button exist in the layout XML? I think you should access it through `binding.addCourseButton` and have all the properties/attributes set there